pyngo/querydict.py

```diff
@@ -107,13 +112,26 @@ def _is_string_like_field(field: FieldInfo) -> bool:
 
 def _is_sequence_field(field: FieldInfo) -> bool:
     """
-    Check if a field is a list field.
+    Check if a field is a sequence field.
 
     Args:
         field (FieldInfo): The field to check.
 
     Returns:
         bool: True if the field is a list field, False otherwise.
     """
+    SEQUENCE_TYPES = (list, tuple, deque, set, frozenset)
     origin_type = get_origin(field.annotation)
-    return origin_type in (list, tuple, deque, set, frozenset)
+    if origin_type in SEQUENCE_TYPES:
+        return True
+    # Hanlde the case of list[int] | None
+    if origin_type is UnionType:
+        union_arg_types = get_args(field.annotation)
+        try:
+            first_member_type, second_member_type = union_arg_types[:2]
+            return (second_member_type is NoneType and get_origin(first_member_type) in SEQUENCE_TYPES) or (
+                first_member_type is NoneType and get_origin(second_member_type) in SEQUENCE_TYPES
+            )
+        except ValueError as e:
+            logger.warning('Error when unpacking union type for "%s" field: %s', field, e)
```

Comment:
```suggestion
            logger.warning(f'Error when unpacking union type for {field} field: {e}')
```

Reply:
@yezz123 Sorry but I have to reject your suggestion on logger, because, your usage means that `field` and `e` will always evaluated to build the f string.
With my usage (passing variables as parameters), the evaluation is delayed until a message needs to be generated (when the log level matches and there is an active handler to write the message out).